Pod/Classes/SEGAmplitudeIntegration.m

```diff
@@ -199,22 +199,18 @@ - (void)reset
 
 #pragma utils
 
-- (void)incrementTrait:(NSString *)trait andValue:(NSString *)value
+- (void)incrementOrSetTraits:(NSDictionary *)traits
 {
-    __block BOOL isAmountSet = false;
-
-    NSArray *increments = self.settings[@"traitsToIncrement"];
-    for (NSString *increment in increments) {
-        if ([increment isEqualToString:trait]) {
+    for (NSString *trait in traits) {
+        id value = [traits valueForKey:trait];
```

Comment:
Is it ok to use `id`?

Reply:
Not sure what the value will be for each `trait`

Reply:
Yes, but is it ok to call `set`/`add` on Amplitude and pass it an `id`?

Reply:
For `add`, they accept only type NSString or NSNumber and handle this check themselves:

```- (AMPIdentify*)add:(NSString*) property value:(NSObject*) value
{
    if ([value isKindOfClass:[NSNumber class]] || [value isKindOfClass:[NSString class]]) {
        [self addToUserProperties:AMP_OP_ADD property:property value:value];
    } else {
        AMPLITUDE_LOG(@"Unsupported value type for ADD operation, expecting NSNumber or NSString");
    }
    return self;
}
```

`set` is a bit more complex, and eventually gets to a check for each type and executes additional logic given the type.

Reply:
So to confirm, we can call them with any type, and internally they handle it properly or bail gracefully if they can't?

If that's the case then we don't need to anything.

Reply:
Yep